### PR TITLE
[Android] Brave News - fixes visible percentage issue

### PR DESCRIPTION
**Pull request by alexsafe**
Uplift of https://github.com/brave/brave-core/pull/13222

Resolves https://github.com/brave/brave-browser/issues/22476
Resolves https://github.com/brave/brave-browser/issues/22470

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 1]
```diff
@@ -402,13 +402,18 @@ protected void onAttachedToWindow() {
...
             if (BraveActivity.getBraveActivity() != null && mIsNewsOn) {
                 new Handler().post(() -> {
                     Tab tab = BraveActivity.getBraveActivity().getActivityTab();
-
                     if (tab != null && tab.getUrl().getSpec() != null
                             && UrlUtilities.isNTPUrl(tab.getUrl().getSpec())) {
+                        // purges display ads on tab change
+                        if (BraveActivity.getBraveActivity().getLastTabId() != tab.getId()) {
+                            mBraveNewsController.onDisplayAdPurgeOrphanedEvents();
+                        }
```
Comment (tmancey) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:413]: Does this cover if a user has n tabs, and restarts the browser?

File: android/java/org/chromium/chrome/browser/app/BraveActivity.java [hunk 2]
```diff
@@ -381,6 +383,14 @@ public void initializeState() {
         BraveSearchEngineUtils.initializeBraveSearchEngineStates(getTabModelSelector());
     }
 
+    public int getLastTabId() {
+        return mLastTabId;
+    }
+
+    public void setLastTabId(int lastTabId) {
```
Comment (tmancey) [inline android/java/org/chromium/chrome/browser/app/BraveActivity.java:390]: Does this get called on browser startup to set the currently active tab?

Comment (alexsafe) [reply android/java/org/chromium/chrome/browser/app/BraveActivity.java:390]: @tmancey yes, once news in on.

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java (hunk 1 above)
Comment (alexsafe) [reply android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:413]: Yes. on restart the tab id gets incremented and it will be different

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.38.x` approved 👍 QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/13222#issuecomment-1127932436.

**Pull request merged**